pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -1559,19 +1560,37 @@ protected void internalSetPersistence(PersistencePolicies persistence) {
 
     private void doUpdatePersistence(PersistencePolicies persistence) {
         try {
-            updatePolicies(namespaceName, policies -> {
-                policies.persistence = persistence;
-                return policies;
-            });
-            log.info("[{}] Successfully updated persistence configuration: namespace={}, map={}", clientAppId(),
-                    namespaceName, jsonMapper().writeValueAsString(persistence));
+            doUpdatePersistenceAsyn(persistence).get(namespaceResources().getOperationTimeoutSec(), TimeUnit.SECONDS);
         } catch (Exception e) {
-            log.error("[{}] Failed to update persistence configuration for namespace {}", clientAppId(), namespaceName,
-                    e);
-            throw new RestException(e);
+            Throwable cause = e.getCause();
+            if (!(cause instanceof RestException)) {
+                throw new RestException(cause);
+            } else {
+                throw (RestException) cause;
+            }
         }
     }
 
+    private CompletableFuture<Void> doUpdatePersistenceAsyn(PersistencePolicies persistence) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        updatePoliciesAsync(namespaceName, policies -> {
```

Comment:
```suggestion
        return updatePoliciesAsync(namespaceName, policies -> {
```
It's better return directly I think :)

Reply:
Thank you for your review. I have revised it